src/otx/core/model/module/segmentation.py

```diff
@@ -29,9 +29,15 @@ def __init__(
         torch_compile: bool,
     ):
         super().__init__(otx_model, optimizer, scheduler, torch_compile)
+        num_classes = otx_model.config.get("decode_head", {}).get("num_classes", None)
```

Comment:
Here's a question: as far as I know, an OTXModel doesn't always hold a config, does it?

Reply:
Hmm, as far as I know, we always have  "config" in our recipes and rely on it heavily. What cases do you mean?